tiatoolbox/annotation/storage.py

```diff
@@ -1071,80 +1072,147 @@ def to_geodict(self) -> Dict[str, Any]:
 
     @staticmethod
     def _dump_cases(
-        fp: Union[IO, str, Path, None],
+        full_path: Union[IO, str, Path, None],
         file_fn: Callable[[IO], None],
         none_fn: Callable[[], Union[str, bytes]],
     ) -> Optional[Union[str, bytes]]:
         """Helper function to handle cases for dumping.
 
         Args:
-            fp:
+            full_path:
                 The file path or handle to dump to.
             file_fn(Callable):
-                The function to call when fp is a file handle.
+                The function to call when full_path is a file handle.
             none_fn(Callable):
-                The function to call when fp is None.
+                The function to call when full_path is None.
 
         Returns:
             Any:
                 The result of dump. Depends on the provided functions.
 
         """
-        if fp is not None:
+        if full_path is not None:
             # It is a file-like object, write to it
-            if hasattr(fp, "write"):
-                return file_fn(fp)
+            if hasattr(full_path, "write"):
+                return file_fn(full_path)
             # Turn a path into a file handle, then write to it
-            with open(fp, "w", encoding="utf-8") as file_handle:
+            with open(full_path, "w", encoding="utf-8") as file_handle:
                 return file_fn(file_handle)
         # Return as str or bytes if no handle/path is given
         return none_fn()
 
     @staticmethod
     def _load_cases(
-        fp: Union[IO, str, Path],
+        full_path: Union[IO, str, Path],
         string_fn: Callable[[Union[str, bytes]], Any],
         file_fn: Callable[[IO], Any],
     ) -> Any:
         with contextlib.suppress(OSError):
-            if isinstance(fp, (Path, str)) and Path(fp).exists():
-                with open(fp) as file_handle:
+            if isinstance(full_path, (Path, str)) and Path(full_path).exists():
+                with open(full_path) as file_handle:
                     return file_fn(file_handle)
-        if isinstance(fp, (str, bytes)):
-            return string_fn(fp)
-        if hasattr(fp, "read"):
-            return file_fn(fp)
+        if isinstance(full_path, (str, bytes)):
+            return string_fn(full_path)
+        if hasattr(full_path, "read"):
+            return file_fn(full_path)
         raise IOError("Invalid file handle or path.")
 
     @classmethod
-    def from_geojson(cls, fp: Union[IO, str]) -> "AnnotationStore":
-        geojson = cls._load_cases(
-            fp=fp,
+    def from_geojson(
+        cls,
+        full_path: Union[IO, str],
+        scale_factor: float = 1,
```

Comment:
Maybe overkill but perhaps list of transforms could be given to allow other things just as inverting the origin / mirroring or rotating etc. Similar to the coordinateTransformations in NGFF v0.4.